tests/jballerina-unit-test/src/test/resources/test-src/types/regexp/regexp_value_test.bal

```diff
@@ -527,6 +527,12 @@ function testExactEqualityWithRegExp() {
     assertEquality(true, re `a\td-*ab[^c-f]+(?m:xj(?i:x|y))` === x4);
 }
 
+function testFreezeDirectWithRegExp() {
+    string:RegExp regExp = re `[^0-9]*`;
+    any result = regExp.cloneReadOnly();
+    assertEquality(true, result is readonly);
+}
```

Comment:
When I checked, this snippet did not reproduce the issue. Hence the `freezeDirect()` method does not get covered. 
Code sample provided in the issue is solved with the changes and hence covers the `freezeDirect()` method. Shall we change the test?

Reply:
Seems we call only `frozenCopy()` not the `freezeDirect()` inside the `cloneReadOnly()` method. I'll add another test.